test/spec/Control/Biegunka/Source/GitSpec.hs

```diff
@@ -0,0 +1,148 @@
+module Control.Biegunka.Source.GitSpec (spec) where
+import           Test.Hspec
+import           Test.Hspec.Expectations.Contrib
+
+import           Control.Biegunka.Execute.Exception   (SourceException (..))
+import qualified Control.Biegunka.Source.Git.Internal as Git
+
+import           Control.Monad                        (liftM)
+import           Data.Maybe                           (listToMaybe)
+import           System.Directory                     (createDirectory)
+import           System.FilePath                      ((</>))
+import           System.IO.Temp                       (withSystemTempDirectory)
+import           System.Random
+
+spec :: Spec
+spec = do
+  around withBiegunkaDirectory $
+    describe "updateGit" $ do
+      context "when local path doesn't exist" $
+        it "creates new directory and sets branch correctly" $ \tmp -> do
+          (repoRemote, repoLocal) <- buildRemoteRepo tmp
+          Git.updateGit repoRemote repoLocal Git.defaultGit
+          currentBranch repoLocal `shouldReturn` Just "master"
+          modifiedFiles repoLocal `shouldReturn` []
+
+      context "when local branch has no commits ahead of remote branch" $ do
+        context "when remote branch has no new commits" $
+          it "does nothing" $ \tmp -> do
```

Comment:
I don't think "does nothing" is a good fit for the test description. How about "does not change checkout state" or something?